tools/tfuser/cmds_provision.go

```diff
@@ -20,99 +16,114 @@ import (
 	"github.com/urfave/cli"
 )
 
-var (
-	day             = time.Hour * 24
-	defaultDuration = day * 30
-)
+func cmdsProvision(c *cli.Context) error {
+	var (
+		seedPath   = mainSeed
+		d          = c.String("duration")
+		assets     = c.StringSlice("asset")
+		volumes    = c.StringSlice("volume")
+		containers = c.StringSlice("container")
+		zdbs       = c.StringSlice("zdb")
+		kubes      = c.StringSlice("kube")
+		networks   = c.StringSlice("network")
+		err        error
+	)
 
-func encryptSecret(plain, nodeID string) (string, error) {
-	if len(plain) == 0 {
-		return "", nil
-	}
+	volumeBuilders := []builders.VolumeBuilder{}
+	for _, vol := range volumes {
+		f, err := os.Open(vol)
+		if err != nil {
+			return errors.Wrap(err, "failed to open volume")
+		}
 
-	pubkey, err := crypto.KeyFromID(pkg.StrIdentifier(nodeID))
-	if err != nil {
-		return "", err
+		volumeBuilder, err := builders.LoadVolumeBuilder(f)
+		if err != nil {
+			return errors.Wrap(err, "failed to load the reservation builder")
+		}
+
+		volumeBuilders = append(volumeBuilders, *volumeBuilder)
 	}
 
-	encrypted, err := crypto.Encrypt([]byte(plain), pubkey)
-	return hex.EncodeToString(encrypted), err
-}
+	containerBuilders := []builders.ContainerBuilder{}
+	for _, cont := range containers {
+		f, err := os.Open(cont)
+		if err != nil {
+			return errors.Wrap(err, "failed to open container")
+		}
 
-func provisionCustomZDB(r *provision.Reservation) error {
-	var config provision.ZDB
-	if err := json.Unmarshal(r.Data, &config); err != nil {
-		return errors.Wrap(err, "failed to load zdb reservation schema")
-	}
+		containerBuilder, err := builders.LoadContainerBuilder(f)
+		if err != nil {
+			return errors.Wrap(err, "failed to load the reservation builder")
+		}
 
-	encrypted, err := encryptSecret(config.Password, r.NodeID)
-	if err != nil {
-		return err
+		containerBuilders = append(containerBuilders, *containerBuilder)
 	}
 
-	config.Password = encrypted
-	r.Data, err = json.Marshal(config)
-
-	return err
-}
+	zdbBuilders := []builders.ZdbBuilder{}
+	for _, zdb := range zdbs {
+		f, err := os.Open(zdb)
+		if err != nil {
+			return errors.Wrap(err, "failed to open zdb")
+		}
 
-func provisionCustomContainer(r *provision.Reservation) error {
-	var config provision.Container
-	var err error
-	if err := json.Unmarshal(r.Data, &config); err != nil {
-		return errors.Wrap(err, "failed to load zdb reservation schema")
-	}
+		zdbBuilder, err := builders.LoadZdbBuilder(f)
+		if err != nil {
+			return errors.Wrap(err, "failed to load the zdb builder")
+		}
 
-	if config.SecretEnv == nil {
-		config.SecretEnv = make(map[string]string)
+		zdbBuilders = append(zdbBuilders, *zdbBuilder)
```

Comment:
Why do u put this in this list first, and not call `reservation.WithZdb` directly ?